### ISSUE TITLE
Failed installation of new release in R version 3.4

### ISSUE THREAD

**Issue by singlecell123** [steinbaugh/basejump#118]
After successful loading of devtools, remotes and GenomeInfoDbData; basejump itself will not install in R version 3.4. Last week I was able to install basejump in the same R version. 

> biocLite("steinbaugh/basejump")
BioC_mirror: https://bioconductor.org
Using Bioconductor 3.6 (BiocInstaller 1.28.0), R 3.4.0 (2017-04-21).
Installing github package(s) ‘steinbaugh/basejump’
Downloading GitHub repo steinbaugh/basejump@master
from URL https://api.github.com/repos/steinbaugh/basejump/zipball/master
Installing basejump
"C:/PROGRA~1/R/R-34~1.0/bin/x64/R" --no-site-file --no-environ --no-save --no-restore --quiet CMD INSTALL  \
  "C:/Users/krkq195/AppData/Local/Temp/RtmpGsTscL/devtools46e0a277bdb/steinbaugh-basejump-d3d215d" --library="C:/Program  \
  Files/R/R-3.4.0/library" --install-tests 

* installing *source* package 'basejump' ...
** R
** data
*** moving datasets to lazyload DB
** inst
** tests
** preparing package for lazy loading
Error in namespaceExport(ns, exports) : 
  undefined exports: %s!=%, %s!==%, %s+%, %s<%, %s<=%, %s==%, %s===%, %s>%, %s>=%, %stri!=%, %stri!==%, %stri+%, %stri<%, %stri<=%, %stri==%, %stri===%, %stri>%, %stri>=%, stri_datetime_add<-, stri_sub<-, stri_subset<-, stri_subset_charclass<-, stri_subset_coll<-, stri_subset_fixed<-, stri_subset_regex<-, stri_c, stri_c_list, stri_cmp, stri_cmp_eq, stri_cmp_equiv, stri_cmp_ge, stri_cmp_gt, stri_cmp_le, stri_cmp_lt, stri_cmp_neq, stri_cmp_nequiv, stri_compare, stri_conv, stri_count, stri_count_boundaries, stri_count_charclass, stri_count_coll, stri_count_fixed, stri_count_regex, stri_count_words, stri_datetime_add, stri_datetime_create, stri_datetime_fields, stri_datetime_format, stri_datetime_fstr, stri_datetime_now, stri_datetime_parse, stri_datetime_symbols, stri_detect, stri_detect_charclass, stri_detect_coll, stri_detect_fixed, stri_detect_regex, stri_dup, stri_duplicated, stri_duplicated_any, stri_enc_detect, stri_enc_detect2, stri_enc_fromutf32, stri_enc_get, stri_enc_info, stri_enc_isascii, stri_enc_isutf16be, stri_enc_isutf16le, stri_enc_isutf32be, stri_enc_isutf32le, stri_enc_isutf8, stri_enc_list, stri_enc_mark, stri_enc_set, stri_enc_toascii, stri_enc_tonative, stri_enc_toutf32, stri_enc_toutf8, stri_encode, stri_endswith, stri_endswith_charclass, stri_endswith_coll, stri_endswith_fixed, stri_escape_unicode, stri_extract, stri_extract_all, stri_extract_all_boundaries, stri_extract_all_charclass, stri_extract_all_coll, stri_extract_all_fixed, stri_extract_all_regex, stri_extract_all_words, stri_extract_first, stri_extract_first_boundaries, stri_extract_first_charclass, stri_extract_first_coll, stri_extract_first_fixed, stri_extract_first_regex, stri_extract_first_words, stri_extract_last, stri_extract_last_boundaries, stri_extract_last_charclass, stri_extract_last_coll, stri_extract_last_fixed, stri_extract_last_regex, stri_extract_last_words, stri_flatten, stri_info, stri_isempty, stri_join, stri_join_list, stri_length, stri_list2matrix, stri_locale_get, stri_locale_info, stri_locale_list, stri_locale_set, stri_locate, stri_locate_all, stri_locate_all_boundaries, stri_locate_all_charclass, stri_locate_all_coll, stri_locate_all_fixed, stri_locate_all_regex, stri_locate_all_words, stri_locate_first, stri_locate_first_boundaries, stri_locate_first_charclass, stri_locate_first_coll, stri_locate_first_fixed, stri_locate_first_regex, stri_locate_first_words, stri_locate_last, stri_locate_last_boundaries, stri_locate_last_charclass, stri_locate_last_coll, stri_locate_last_fixed, stri_locate_last_regex, stri_locate_last_words, stri_match, stri_match_all, stri_match_all_regex, stri_match_first, stri_match_first_regex, stri_match_last, stri_match_last_regex, stri_na2empty, stri_numbytes, stri_opts_brkiter, stri_opts_collator, stri_opts_fixed, stri_opts_regex, stri_order, stri_pad, stri_pad_both, stri_pad_left, stri_pad_right, stri_paste, stri_paste_list, stri_rand_lipsum, stri_rand_shuffle, stri_rand_strings, stri_read_lines, stri_read_raw, stri_remove_empty, stri_replace, stri_replace_all, stri_replace_all_charclass, stri_replace_all_coll, stri_replace_all_fixed, stri_replace_all_regex, stri_replace_first, stri_replace_first_charclass, stri_replace_first_coll, stri_replace_first_fixed, stri_replace_first_regex, stri_replace_last, stri_replace_last_charclass, stri_replace_last_coll, stri_replace_last_fixed, stri_replace_last_regex, stri_replace_na, stri_reverse, stri_sort, stri_split, stri_split_boundaries, stri_split_charclass, stri_split_coll, stri_split_fixed, stri_split_lines, stri_split_lines1, stri_split_regex, stri_startswith, stri_startswith_charclass, stri_startswith_coll, stri_startswith_fixed, stri_stats_general, stri_stats_latex, stri_sub, stri_subset, stri_subset_charclass, stri_subset_coll, stri_subset_fixed, stri_subset_regex, stri_timezone_get, stri_timezone_info, stri_timezone_list, stri_timezone_set, stri_trans_char, stri_trans_general, stri_trans_isnfc, stri_trans_isnfd, stri_trans_isnfkc, stri_trans_isnfkc_casefold, stri_trans_isnfkd, stri_trans_list, stri_trans_nfc, stri_trans_nfd, stri_trans_nfkc, stri_trans_nfkc_casefold, stri_trans_nfkd, stri_trans_tolower, stri_trans_totitle, stri_trans_toupper, stri_trim, stri_trim_both, stri_trim_left, stri_trim_right, stri_unescape_unicode, stri_unique, stri_width, stri_wrap, stri_write_lines
ERROR: lazy loading failed for package 'basejump'
* removing 'C:/Program Files/R/R-3.4.0/library/basejump'
Installation failed: Command failed (1)
Error in if (file.exists(dest) && file.mtime(dest) > file.mtime(lib) &&  : 
  missing value where TRUE/FALSE needed

**Comment by mjsteinbaugh** [comment]
Thanks @singlecell123 I do test the basejump code base against Windows, so this should install correctly. It looks like there may be an issue with the stringi package from CRAN installating correctly:
https://cran.r-project.org/web/packages/stringi/index.html
Maybe try installing that package manually first to see if it works.

**Comment by singlecell123** [comment]
Thanks for your quick response! I have re-installed stringi both locally and from CRAN and it doesn’t seem to make any difference as I can load it fine, but then when I try to install basejump from github (or Bioconductor or locally) I get the error message: “Installation failed: missing value where TRUE/FALSE needed”. I’m not using R studio as I thought that might be part of the problem..

**Comment by mjsteinbaugh** [comment]
Hi @singlecell123, it appears that there's a problem with your R installation. Note that Windows requires R to be run in Administrator mode to install packages, unless you are installing to a user library. By default, R on Windows installs into a system library inside `C:/Program 
Files/R/R-3.4.0/library` (see error log above). This can be configured by setting `Renviron` – see the [Startup](https://stat.ethz.ch/R-manual/R-devel/library/base/html/Startup.html) documentation in the R reference manual for more information. I'd recommend creating a user library outside of `Program Files`, and trying the install again from scratch to see if the error is reproducible.

**Comment by singlecell123** [comment]
Hi I have just managed to solve the problem – installing basejump through Bioconductor was calling the function install.packages() in my R version 3.4, which introduced a bug due to an NA value being save in m.time(lib). Apparently this is a bug that is part of R 3.4 and is has been fixed in later versions. I just had to restart R as to remove this bug and then install basejump through github instead. Thanks for the help!
